spec/classes/application_spec.rb

```diff
@@ -280,6 +286,96 @@ class {'katello::globals':
           it { is_expected.to create_package('rubygem-katello').that_requires('Anchor[katello::candlepin]') }
         end
       end
+
+      context 'with katello::params::enable_katello_agent true' do
+        let :pre_condition do
+          <<-EOS
+          class {'katello::params':
+            enable_katello_agent => true,
+            candlepin_oauth_secret => 'candlepin-secret',
+          }
+          EOS
+        end
+
+        it { is_expected.to compile.with_all_deps }
+
+        let(:katello_yaml_content) do
+          [
+            ':katello:',
+            '  :rest_client_timeout: 3600',
+            '  :content_types:',
+            '    :yum: true',
+            '    :file: true',
+            '    :deb: true',
+            '    :puppet: false',
+            '    :docker: true',
+            '    :ostree: false',
+            '  :candlepin:',
+            '    :url: https://localhost:23443/candlepin',
+            '    :oauth_key: "katello"',
+            '    :oauth_secret: "candlepin-secret"',
+            '    :ca_cert_file: /etc/pki/katello/certs/katello-default-ca.crt',
+            '  :candlepin_events:',
+            '    :ssl_cert_file: /etc/pki/katello/certs/java-client.crt',
+            '    :ssl_key_file: /etc/pki/katello/private/java-client.key',
+            '    :ssl_ca_file: /etc/pki/katello/certs/katello-default-ca.crt',
+            '  :agent:',
+            '    :enabled: true',
+            '    :broker_url: amqps://localhost:5671',
+            '    :event_queue_name: katello.agent',
+            '  :katello_applicability: true',
+          ]
+        end
+
+        it 'should generate correct katello.yaml' do
+          verify_exact_contents(catalogue, '/etc/foreman/plugins/katello.yaml', katello_yaml_content)
+        end
+      end
+
+      context 'with katello::params::enable_katello_agent empty string' do
```

Comment:
@ehelms could you explain why this empty string test is needed?

Reply:
So this becomes `$enable_katello_agent_real => false`. Ok :+1:

Reply:
And `katello::params::enable_katello_agent` specifically supports the data types `Variant[Boolean, Enum['']]`

Reply:
So what is the benefit of allowing both `$enable_katello_agent => false` and `$enable_katello_agent => ''` and making them equivalent?

Reply:
The explanation I put into the params doc is:

```
# @param enable_katello_agent
#   Set to true to setup Qpid and katello-agent infrastructure. Accepts
#   empty string value to allow Hiera configuration of the value.
```

Does that answer your question or could I word this differently to convey it better?